src/consensus/ledger_enclave.h

```diff
@@ -103,5 +89,15 @@ namespace consensus
     {
       RINGBUFFER_WRITE_MESSAGE(consensus::ledger_truncate, to_host, idx);
     }
+
+    /**
+     * Compact the ledger at a given index.
+     *
+     * @param idx Index to compact at
+     */
+    void compact(Index idx)
```

Comment:
Calling this compact isn't right. We compact the store, but we `freeze` the ledger, or we `make_immutable`, or `advance_read_only_watermark`, possibly `finalise()`.

At worst, `commit()`.

Reply:
I went for `commit()` to stay consistent with the rest of the interface. It doesn't necessarily mean that the ledger is frozen at that point though, it just notifies the ledger that the consensus has agreed on an index. What it guarantees though is that all indices included in _complete_ chunks that are `<=` the commit idx will be committed and forever closed for writing. I hope this is clearer now but let me know if it isn't.

Reply:
I think it logically means that entries in the ledger up to that point cannot ever be rolled back or modified, even if they are part of a file that still has a mutable suffix. I'm fine with commit(), but compact() was inaccurate.